New reference architecture diagram - Optimizing device roaming experience with geolocated IPs (#18735)
cloudflare/cloudflare-docs: src/content/docs/reference-architecture/diagrams/network/optimizing-roaming-experience-with-geolocated-ips.mdx

---
title: Optimizing device roaming experience with geolocated IPs
pcx_content_type: reference-architecture-diagram
products:
  - Magic WAN
  - Gateway
sidebar:
  order: 1
  label: Device roaming with geolocated IPs
updated: 2024-12-13
---

## Introduction

> A private APN enables devices, like connected vehicles, connected containers, healthcare devices or drones, to be connected while roaming across different countries. The device connects with a SIM or eSIM card to a dedicated network and as the device moves to a new country, it automatically selects the appropriate private APN for the local provider.

Reviewer: ```suggestion
A private [Access Point Name](https://en.wikipedia.org/wiki/Access_Point_Name) (APN) enables devices, like connected vehicles, connected containers, healthcare devices or drones, to be connected while roaming across different countries. The device connects with a SIM or eSIM card to a dedicated network, and as the device moves to a new country, it automatically selects the appropriate private APN for the local provider.
```

> APN traffic, typically managed by a third party provider such as a telecommunications company, is routed through specific regional Internet breakouts to get access to the Internet. This architecture can create challenges in regards to the localization of that traffic. For example, a device roaming in France, might have traffic exit to the Internet from a UK based Internet breakout. Therefore web sites and other Internet services will treat the device as if it is in the UK, and deliver content in the wrong language or apply regional restrictions.

Reviewer: ```suggestion
APN traffic, typically managed by a third party provider such as a telecommunications company, is routed through specific regional Internet breakouts to get access to the Internet. This architecture can create challenges in regards to the localization of that traffic. For example, a device roaming in France might have traffic exit to the Internet from a UK-based Internet breakout. Therefore web sites and other Internet services will treat the device as if it is in the UK and deliver content in the wrong language or apply regional restrictions.
```

> In this document, we'll discuss how Cloudflare can be used to solve this problem and we will use the example of a service provider using private mobile networks [Access Point Names](https://en.wikipedia.org/wiki/Access_Point_Name) (APN) to connect devices roaming across multiple countries through regional Internet breakouts. This use case is relevant to global enterprises with regional offices, transportation fleets with connected vehicles, or any organization needing to maintain consistent, secure, and region-specific connectivity for roaming devices.

Reviewer: ```suggestion
In this document, we'll discuss how Cloudflare can be used to solve this problem and will use the example of a service provider using private mobile networks (APNs) to connect devices roaming across multiple countries through regional Internet breakouts. This use case is relevant to global enterprises with regional offices, transportation fleets with connected vehicles, or any organization needing to maintain consistent, secure, and region-specific connectivity for roaming devices.
```

![Figure 1: Showing how Internet breakouts can present an egress IP that doesn't match the country the device is in.](~/assets/images/reference-architecture/optimizing-roaming-experience-with-geolocated-ips/figure1.svg "Figure 1: Showing how Internet breakouts can present an egress IP that doesn't match the country the device is in.")

# Correctly locate and secure devices by connecting them to the Cloudflare global network

> Cloudflare addresses these challenges by routing device traffic from the Internet breakout to our global network, where traffic is processed at a Cloudflare data center close to the Internet breakout. This allows for two benefits.

Reviewer: ```suggestion
Cloudflare addresses these challenges by routing device traffic from the Internet breakout to our global network, where traffic is processed at a Cloudflare data center close to the Internet breakout. This allows for two benefits:
```

1. Cloudflare can analyse the traffic, determine the original country of origin, and then ensure that traffic egresses onto the Internet from an IP address that is geolocated to the same country of origin.

> 2. Cloudflare can filter traffic based on [secure web gateway](/cloudflare-one/policies/gateway/) policies, allowing you to protect devices from access risky Internet hosts. It also allows you to lock down access for devices to specific Internet hosts, such as only allow devices to make requests to APIs that support their function.

Reviewer: ```suggestion
2. Cloudflare can filter traffic based on [secure web gateway](/cloudflare-one/policies/gateway/) policies, allowing you to protect devices from accessing risky Internet hosts. It also allows you to lock down access for devices to specific Internet hosts, such as only allow devices to make requests to APIs that support their function.
```
